package.json

```diff
@@ -43,6 +43,7 @@
     "eslint-plugin-prettier": "^3.4.0",
     "eslint-plugin-react": "^7.19.0",
     "eslint-plugin-react-hooks": "^3.0.0",
+    "ethereum-waffle": "^3.4.0",
```

Comment:
Although technically `ethereum-waffle` is a full Ethereum (compile, deploy, test) toolkit, we import it here because it actually adds some great ["chai matchers"](https://ethereum-waffle.readthedocs.io/en/latest/matchers.html#) which make tests really readable. I added it here for demonstration purposes because the [optimism-tutorial](https://github.com/OpenZeppelin/openzeppelin-contracts) uses as an auxillary package to `hardhat` which they use to compile and deploy.

Reply:
Question is whether we should adopt `waffle` into our repo, so this PR demonstrates its potential usage.

Reply:
Does this mean we need to use ethers wherever we use waffle?

If so, does this mean we need to support ethers everywhere? Not sure how much overhead it would add to trying to support both ethers and web3.

Reply:
Yeah it seems waffle + ethers are linked. I think i'm going to remove waffle and just use web3